### DISCUSSION TITLE
Typography extend screen modifiers

### DISCUSSION THREAD

**Discussion by stephenturvey** [tailwindlabs/tailwindcss#4119] [Help]
I'm having trouble extending the typography plugin to specify my heading font sizes for `md`, `lg` etc modifiers. Values under the `DEFAULT` key work fine, but the following declarations are not recognised (no error but don't produce any css).

```
extend: {
      typography: {
        // works fine
        DEFAULT: {
          css: {
            h2: {fontSize: '0.75rem'},
          },
        },
        // not recognised
        md: {
          css: {
            h2: {fontSize: '1.5rem'},
          },
        },
        // not recognised
        lg: {
          css: {
            h2: {fontSize: '1.875rem'},
          },
        },
      },
},

**Comment by simonswiss** [comment]
Hey!

Worth noting that targetting these modifiers will not automatically change the typographic styles at different breakpoints. What they do is change the styles applied when you use classes like `prose-md` or `prose-lg` for specific breakpoints.

Here's an example where I change the size/color of `p` tags for the typography plugin.

For it to work, I need to apply the following classes:

```html<div class="mx-auto prose md:prose-md lg:prose-lg">```

Here's a link to a Tailwind Play, based of your code snippet above:

https://play.tailwindcss.com/ukRlDebiXG

Hope it helps!

**Reply by stephenturvey** [reply]
Thanks Simon, I understand that now. What would be the best way for me to achieve what I want - i.e. changing the heading sizes responsively but for the `prose` class only? 

I want virtually all text to be the same size at all breakpoints hence only using the one `prose` class on the container. I can't think of anything other than headings I want to modify.

**Comment by bakura10** [comment]
Hi !

I found the exact same issue. My prose are styled consistently so this feels a bit redundant to add "prose md:prose-md lg:prose-lg". I actually forgot to add it a few times, just leaving the standard "prose". Is there a way to override the default values per breakpoint @simonswiss ?
